portfolio/src/main/webapp/script.js

```diff
@@ -174,3 +183,53 @@ function getImageUrls() {
       });
 }
 
+/**
+ * Gets the user's authentication status from the auth-status servlet
+ * and calls handleAuthenticationStatus function to change form visibility
+ * accordingly
+ */
+function checkAuthentication() {
+  fetch('/auth-status')
+      .then((response) => response.json())
+      .then((authInfo) => {
+        // Get the authentication status as boolean and auth link as string
+        const loggedIn = authInfo['logged-in'];
+        const authLink = authInfo['link'];
+
+        // Begin changes to the DOM depending on the authentication status
+        initializeFormState(loggedIn, authLink);
+      });
+}
+
+/**
+ * Updates the HTML elements to prompt the user to log in or log out,
+ * with the correct link to do so below the text. Hides the comment
+ * form if logged out, shows it if logged in.
+ * @param {boolean} loggedIn true if user is logged in, false otherwise
+ * @param {string} authLink link to either log in or log out, as needed.
+ */
+function initializeFormState(loggedIn, authLink) {
+  // Get the HTML elements
+  const authPromptElement =
+      document.getElementById('comment-form-authentication-prompt');
+
+  const authLinkElement =
+      document.getElementById('comment-form-authentication-link');
+
+  const form = document.getElementById('comment-form');
+
+  // If logged in, prompt user with link to log out and show form
+  // If logged out, prompt user to log in and hide form
+  if (loggedIn) {
+    authPromptElement.innerText = logOutPrompt;
+    authLinkElement.innerHTML = 'Log Out of Google Account';
```

Comment:
For consistency, either have both constants inline or put both in variables.

Alternatively, you have separate elements loginPrompt/loginLink and logoutPrompt/logoutLink. That way most of the presentation (the text) can stay in the HTML and you can show/hide the correct one.

Tip: look into `querySelector` to find sub-elements of a target element. https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector

Reply:
I actually tried using the hidden elements now - looks cool! 

In terms of the query selector, what are the advantages to using this vs. Just using getElementById? I feel like that's a simple enough method to access elements in the DOM.

Reply:
Yeah, this is much cleaner! 

`getElementById` is fine here. I was just suggesting `querySelector` as a more powerful tool, because it supports full CSS selectors and allows searching for descendants of an element other than `<document>`.